### PR TITLE
Search filter screen

### PR DESCRIPTION
**Pull request by EmmanuelWLX**
## Summary
- Added SearchScreen, which is reached with the lens icon in BookList screen
- Created SearchBar component to achieve this goal

### Android
search-nexus6p

### iOS
search-iphone6

## Trello card
https://trello.com/c/mSBFolJo/34-n-104-filtro-de-b%C3%BAsqueda

### CODE REVIEW

File: LibraryApp/src/app/screens/SearchScreen/index.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  compareBookTitle = (bookA, bookB) => {
+    if (bookA.title < bookB.title) return -1;
+    if (bookB.title < bookA.title) return 1;
+    return 0;
+  };
+
+  handleSearch = value => {
+    const { books } = this.props;
+    const booksFiltered = Immutable.asMutable(books.filter(this.searchTitle(value))).sort(
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/screens/SearchScreen/index.js]: You must find a way to do this without the Immutable

File: LibraryApp/src/app/screens/SearchScreen/components/EmptySearch/index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import { View, Text, Image } from 'react-native';
+
+import searchIcon from './assets/ic_search@3x.png.png';
+import styles from './styles';
+
+function EmptySearch() {
+  return (
+    <View style={styles.container}>
+      <Image source={searchIcon} style={styles.searchIcon} />
+      <Text style={styles.searchTitle}>Search in Wbooks</Text>
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/screens/SearchScreen/components/EmptySearch/index.js]: Remember to create a `strings.js` file for this kind of texts

File: LibraryApp/src/app/components/SearchBar/styles.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  container: {
+    alignItems: 'center',
+    backgroundColor: white,
+    borderRadius: BAR_BORDER_RADIUS,
+    flexDirection: 'row',
+    height: BAR_HEIGHT,
+    paddingHorizontal: BAR_PADDING_HORIZONTAL
+  },
+  containerInput: {
+    width: '90%'
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/components/SearchBar/styles.js]: Try to use margin instead

File: LibraryApp/src/app/screens/SearchScreen/index.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      year: PropTypes.string
+    }).isRequired
+  )
+};
+
+const mapStateToProps = store => ({
+  books: store.books.books
+});
+
+const SearchScreenConnected = connect(mapStateToProps)(SearchScreenContainer);
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/screens/SearchScreen/index.js]: We dont use an object in cases with just `connect(mapStateToProps, mapDispatchToProps)`, just single line in export

File: LibraryApp/src/app/screens/SearchScreen/index.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    );
+  }
+}
+
+SearchScreenContainer.propTypes = {
+  navigation: PropTypes.shape({
+    dispatch: PropTypes.func
+  }),
+  books: PropTypes.arrayOf(
+    PropTypes.shape({
```
Comment (mmamaniwolox) [inline LibraryApp/src/app/screens/SearchScreen/index.js]: For this you can create a `book.js` file in a `propTypes` folder and export this object you may use somewhere else

File: LibraryApp/src/app/screens/SearchScreen/index.js (hunk 4 above)
Comment (EmmanuelWLX) [reply LibraryApp/src/app/screens/SearchScreen/index.js]: that is true, i had forgotten that i discovered another way to apply the navigationOptions without it

**Pull request merged**